### PR TITLE
REMReM should read the Jasypt encryption key from jasypt.key file.

### PR DESCRIPTION
**Pull request by durga-vasaadi**
### Description of the Change
REMReM publish able to read the Jasypt encryption key from jasypt.key file.

### Benefits
REMReM publish able to read the Jasypt encryption key from a separate file.
The Jasypt encrypted key should place in the functional user path like
<path>/jasypt.key

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 1]
```diff
@@ -77,6 +80,8 @@
     private void readSpringProperties() {
         JsonNode rabbitmqInstancesJsonListJsonArray = null;
         final ObjectMapper objMapper = new ObjectMapper();
+        final String jasyptKey = readJasyptKeyFile(jasyptPassword);
```
Comment (raja-maragani) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:83]: jasyptPassword is password or a jasypt key file path?

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 2]
```diff
@@ -104,6 +109,23 @@ private void readSpringProperties() {
...
+    //it read and store the jasypt key from jasypt.key file
+    public static String readJasyptKeyFile(String jasyptPassword) {
+        String jasyptKey = "";
+        try {
+
+            if (StringUtils.isNotBlank(jasyptPassword)) {
+                jasyptKey = new String(Files.readAllBytes(Paths.get(jasyptPassword)));
+            } else {
+                return jasyptKey;
+            }
```
Comment (raja-maragani) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:121]: remove this code, no use with these lines

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java (hunk 1 above)
Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:83]: jasyptPassword value is jasyptKey file path, so do I change variable to jasyptKeyFilePath?
I have question regarding property value of jasypt.encryptor.password in application.properties. Now we are passing the jasypt.encryptor.password value is jasypt key file path. So can I change this property name to jasypt.encryptor.jasyptKeyFilePath or something?

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java (hunk 2 above)
Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:119]: removed else part code

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java (hunk 1 above)
Comment (raja-maragani) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:83]: yes please change. jasyptKeyFilePath is fine.

Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Changed to jasyptKeyFilePath in application.properties and code.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 3]
```diff
@@ -104,6 +109,21 @@ private void readSpringProperties() {
...
+    //it read and store the jasypt key from jasypt.key file
+    public static String readJasyptKeyFile(String jasyptKeyFilePath) {
+        String jasyptKey = "";
+        try {
+
+            if (StringUtils.isNotBlank(jasyptKeyFilePath)) {
+                jasyptKey = new String(Files.readAllBytes(Paths.get(jasyptKeyFilePath)));
+            }
+
+        } catch (IOException e) {
```
Comment (xumakap) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:121]: Print exception message

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 4]
```diff
@@ -104,6 +109,21 @@ private void readSpringProperties() {
...
+            System.out.println("Could not read the jasypt key from " + jasyptKeyFilePath);
```
Comment (xumakap) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:122]: Use logger

Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: done

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java (hunk 3 above)
Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:121]: done

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 5]
```diff
@@ -119,7 +119,7 @@ public static String readJasyptKeyFile(String jasyptKeyFilePath) {
             }
 
         } catch (IOException e) {
-            System.out.println("Could not read the jasypt key from " + jasyptKeyFilePath);
+            log.error("Could not read the jasypt key from the jasypt key file path: " + e.getMessage(), e);
```
Comment (xumakap) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:122]: Why printing both exception.getMessage and exception message? print any one

Comment (zvsddrg) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java:122]: done the change as you suggested

**Pull request merged**